app/containers/Button/index.tsx

```diff
@@ -1,25 +1,21 @@
 import React from 'react';
-import { StyleSheet, Text } from 'react-native';
-import Touchable from 'react-native-platform-touchable';
+import { StyleProp, StyleSheet, Text, TextStyle } from 'react-native';
+import Touchable, { PlatformTouchableProps } from 'react-native-platform-touchable';
 
-import { TSupportedThemes } from '../../theme';
+import { useTheme } from '../../theme';
 import { themes } from '../../lib/constants';
 import sharedStyles from '../../views/Styles';
 import ActivityIndicator from '../ActivityIndicator';
 
-interface IButtonProps {
+interface IButtonProps extends PlatformTouchableProps {
 	title: string;
-	type: string;
-	onPress(): void;
-	disabled: boolean;
-	backgroundColor: string;
-	loading: boolean;
-	theme: TSupportedThemes;
-	color: string;
-	fontSize: any;
-	style: any;
-	styleText?: any;
-	testID: string;
+	onPress: () => void;
```

Comment:
Use `Function` to create a generic.

Reply:
Using `Function` envokes this error `Interface 'IButtonProps' incorrectly extends interface 'PlatformTouchableProps'.
  Types of property 'onPress' are incompatible.
    Type 'Function' is not assignable to type '(event: GestureResponderEvent) => void'.`
Am I missing something?